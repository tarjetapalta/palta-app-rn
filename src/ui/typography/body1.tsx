import React, { FunctionComponent } from 'react';

import { Text, TextProps } from './base';

export const Body1: FunctionComponent<
  Pick<
    TextProps,
    'testID' | 'color' | 'emphasis' | 'textAlign' | 'numberOfLines'
  >
> = ({
  /** Test id for test */
  testID = 'Body1',
  /** Text */
  children,
  /** Text color */
  color,
  /** Text font opacity */
  emphasis,
  /** Text align */
  textAlign,
  /** Number of line to truncate text */
  numberOfLines,
}) => (
  <Text
    testID={testID}
    color={color}
    size="s"
    weight="light"
    emphasis={emphasis}
    textAlign={textAlign}
    numberOfLines={numberOfLines}
  >
    {children}
  </Text>
);
