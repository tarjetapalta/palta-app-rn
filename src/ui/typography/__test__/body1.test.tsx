import React from 'react';
import renderer from 'react-test-renderer';

import WithTheme from '../../../theme/context';
import { Body1 } from '../body1';

describe('body1 component', () => {
  it('basic Body1 with color "text", emphasis "high" and testID "Body1"', () => {
    const result = renderer
      .create(
        <WithTheme>
          <Body1>Some text</Body1>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
  it('basic Body1 with color "primary", emphasis "medium"', () => {
    const result = renderer
      .create(
        <WithTheme>
          <Body1 color="primary" emphasis="medium">
            Some text
          </Body1>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
  it('basic with custom testId', () => {
    const result = renderer
      .create(
        <WithTheme>
          <Body1 testID="custom testID">Some text</Body1>
        </WithTheme>,
      )
      .toJSON();

    expect(result).toMatchSnapshot();
  });
});
