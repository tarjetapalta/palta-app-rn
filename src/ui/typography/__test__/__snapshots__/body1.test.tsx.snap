// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`body1 component basic Body1 with color "primary", emphasis "medium" 1`] = `
<Text
  color="primary"
  emphasis="medium"
  numberOfLines={0}
  size="s"
  style={
    Array [
      Object {
        "color": "#91b086",
        "fontFamily": "BwModelica-Light",
        "fontSize": 14,
        "opacity": 0.6,
        "textAlign": "center",
      },
    ]
  }
  testID="Body1"
  weight="light"
>
  Some text
</Text>
`;

exports[`body1 component basic Body1 with color "text", emphasis "high" and testID "Body1" 1`] = `
<Text
  numberOfLines={0}
  size="s"
  style={
    Array [
      Object {
        "color": "#283238",
        "fontFamily": "BwModelica-Light",
        "fontSize": 14,
        "opacity": 0.87,
        "textAlign": "center",
      },
    ]
  }
  testID="Body1"
  weight="light"
>
  Some text
</Text>
`;

exports[`body1 component basic with custom testId 1`] = `
<Text
  numberOfLines={0}
  size="s"
  style={
    Array [
      Object {
        "color": "#283238",
        "fontFamily": "BwModelica-Light",
        "fontSize": 14,
        "opacity": 0.87,
        "textAlign": "center",
      },
    ]
  }
  testID="custom testID"
  weight="light"
>
  Some text
</Text>
`;
